xbmc/windows/GUIWindowSystemInfo.cpp

```diff
@@ -258,3 +257,13 @@ void CGUIWindowSystemInfo::SetControlLabel(int id, const char *format, int label
       CServiceBroker::GetGUI()->GetInfoManager().GetLabel(info, INFO::DEFAULT_CONTEXT));
   SET_CONTROL_LABEL(id, tmpStr);
 }
+
+void CGUIWindowSystemInfo::LoadPrivacyPolicy()
+{
+  if (!m_privacyPolicyLoaded)
```

Comment:
This will still block the GUI in the first iteration. Maybe you can consider adding this info to `CSysInfoJob` to make it async?

Reply:
Doing it fully async would certainly be better, and I experimented with such an approach, but doesn't the label have to be set in the GUI thread regardless? I tried setting the label with a threaded message on window open, and even though it was dispatched async it still stalled the GUI thread. My best thought would be to add a 1s delay so that it at least loads when there's little activity on the screen, but that seems hacky. What do you think?

Reply:
If memory serves me well a placeholder label is set on the controller while the label is still unavailable (something like "loading" or similar). Check the network tab quick enough right after starting Kodi

Reply:
The problem isn't reading the file, the label is always available instantly. The delay happens when the text is set for the textbox in the GUI thread, as that's where gui messages are handled.

Reply:
Ah right, then I am out of ideas :)
Your approach here is better that what we current have, so +1